File: solutions/python/bob/1/bob.py
def response(hey_bob):
    
    stripped = hey_bob.strip()
    
   
    if not stripped:
        return "Fine. Be that way!"
    
    
    is_question = stripped.endswith("?")
    is_yelling = stripped.isupper()
    
    if is_question and is_yelling:
        return "Calm down, I know what I'm doing!"
    elif is_question:
        return "Sure."
    elif is_yelling:
        return "Whoa, chill out!"
    else:
        return "Whatever."

File: solutions/csharp/bob/1/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        // Conditions: 
        bool endWithQuestionMark = statement.Trim().EndsWith("?");
        bool allCapitalLetters = (statement.Any(char.IsLetter)) && (statement == statement.ToUpper());
        bool allCapitalQuestion = allCapitalLetters && endWithQuestionMark;
        bool silentOrWhitespace = string.IsNullOrWhiteSpace(statement);

        if (allCapitalQuestion)
        {
            return "Calm down, I know what I'm doing!";
        } else if (endWithQuestionMark)
        {
            return "Sure.";
        } else if (allCapitalLetters)
        {
            return "Whoa, chill out!";
        } else if (silentOrWhitespace)
        {
            return "Fine. Be that way!";
        } else 
        {
            return "Whatever.";
        }
       
        
    }
}